### ISSUE TITLE
Unhandled abstract form issue with fun((...) 

### ISSUE THREAD

**Issue by rvangraan** [inaka/elvis#239]
this spec:
```-spec module_map(state(), abstract_function()) -> fun((...) -> ok | {error,term()}).```

causes these errors:
```
Error: '{unhandled_abstract_form,
                           {type,[{line,114},{column,55},
                                   {text,"("}],any}}' while applying rule 'no_spec_with_records'.
 Error: '{unhandled_abstract_form,
                           {type,[{line,114},{column,55},
                                  {text,"("}],any}}' while applying rule 'state_record_and_type'.
```

it is a valid syntax for specs `fun((...) -> ` means a fun of any arity returning a specific type.
```
Fun :: fun()                  %% any function
       | fun((...) -> Type)     %% any arity, returning Type
       | fun(() -> Type)
       | fun((TList) -> Type)
```

**Comment by jfacorro** [comment]
@rvangraan The `master` branch includes this fix, could you please confirm when you get the chance? This fix will be included in the next `0.2.5` release.

**Comment by rvangraan** [comment]
Thanks @jfacorro, I will give it a try.
